integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/impl/AnswersSpaceActivityPublisher.java

```diff
@@ -155,11 +155,10 @@ public void saveAnswer(String questionId, Answer answer, boolean isNew) {
       //
       String answerContent = ActivityUtils.processContent(answer.getResponses());
       
-      if (activityId != null) {
+      if (activityId != null && answer.getApprovedAnswers() == true) {
```

Comment:
You can remove `== true` in the condition.
Other small improvement: the method `getApprovedAnswers()` seems not properly named, `isApproved()` would be clearer.